doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,180 @@
+- `queryName`, the name of the QueryType to run. Example: `AcmeBundle:LatestArticles'
+- `viewType`, similar to what `viewContent` and `viewLocation` expect. Examples: `list`, `tree`, `full_list`...
+
+If the QueryType supports any, they can be provided using the `parameters` argument (hash):
+
+```twig
+{{ render( controller(
+    'ez_query:locationQuery',
+    {
+        'queryName': 'AcmeBundle:LatestContent',
+        'viewType': 'list',
+        parameters: { ContentType: 'article' }
+)) }}
+```
+
+### <a name="template_override"></a>Rendering of queries
+The `ez_query` actions support template override. They support several matchers.
+- `QueryType\Name`: matches the QueryType's name
+- `QueryType\Parameter`: matches the value of one or several of the parameters passed to the QueryType
+- `QueryType\Expression`: matches the parameters with an [Expression](http://symfony.com/en/doc/current/components/expression_language/index.html).
+
+Assuming we want to render the `AcmeBundle:LatestContent` query using
+`AcmeBundle:query/list/latest_articles.html.twig` when the `ContentType` parameter has the value 'article':
+
+```twig
+system:
+    default:
+        query:
+            list:
+                latest_articles:
+                    template: "AcmeBundle:query/list/latest_articles.html.twig"
+                    match:
+                        QueryType\Name: 'AcmeBundle:LatestContent'
+                        QueryType\Parameters: {ContentType: "article"}
+```
+
+#### Template matchers
+
+##### `QueryType\Name`
+Matches the QueryType's name (exact match).
+
+##### `QueryType\Parameters`
+Matches the parameters against a given hash. The hash must contain the parameter's name as the key,
+and the match value as the value:
+
+```twig
+match:
+  QueryType\Parameters:
+    type: "article",
+    category: "development"
+
+Each parameter will be matched exactly.
+The matcher will match if ALL of the provided parameters match what is contained in the hash.
+
+##### `QueryType\Expression`
+Uses Symfony's [Expression Language](http://symfony.com/en/doc/current/components/expression_language/index.html)
+for advanced matching.
+This matcher expects a valid expression language string as the input. The expression must be
+evaluated to a boolean. The parameters hash is available as `parameters`
+
+In the example below, 'type' must be either 'article' or 'blog_post'.
+
+```twig
+match:
+  QueryType\Expression: "parameters['type'] in ['article', 'blog_post']"
+```
+
+### Query controller templates
+The templates used by this controller are mostly meant as container for more sub-requests. Except for very simple cases
+(bullet points, table), you should use a sub-request to the Content controller on the query results.
+
+#### Location Query
+The query results are available as `location_array`, an array of `Location` objects.
+
+#### Content Query
+The Query results are available as `content_array`, an array of ContentInfo objects.
```

Comment:
What about the number of results? Shouldn't we return the entire result set instead?